### DISCUSSION TITLE
Wait for user to connect wallet successfully.

### DISCUSSION THREAD

**Discussion by DinhKhai0201** [WalletConnect/walletconnect-monorepo#4367] [Developer Support] [answered]
Hi,

Currently, I'm using Web3modal.
I want to show a popup immediately after the user successfully connects their wallet, prompting them to sign a message.
Is there a way to wait for the user to connect their wallet successfully, similar to how it worked in Web3modal v1?

Example about Web3modal v1?
```javascript
const web3Modal = new Web3Modal({
    providerOptions // required
});
const provider = await web3Modal.connect();
const web3 = new Web3(provider);

const [address] = await web3.eth.getAccounts();

const signature = await web3.eth.personal.sign("Message", address, '');
```

**Comment by glitch-txs** [comment, marked as answer]
Web3Modal v1 doesn't work any more, it was deprecated. Check: https://docs.walletconnect.com/web3modal/about
